app/models/dialog_field_drop_down_list.rb

```diff
@@ -3,7 +3,53 @@ def show_refresh_button?
     !!show_refresh_button
   end
 
+  def multi_value?
+    return true if options[:force_multi_value].present? &&
+                   options[:force_multi_value] != "null" &&
+                   options[:force_multi_value]
+  end
+
+  def force_multi_value=(setting)
+    options[:force_multi_value] = setting
+  end
+
   def initial_values
     [[nil, "<None>"]]
   end
+
+  def refresh_json_value(checked_value)
+    @raw_values = @default_value = nil
+
+    refreshed_values = values
+
+    @value = if checked_value.kind_of?(Array) &&
```

Comment:
The new additions to this logic block should be tested in the related spec file.